precompile/README.md

```diff
@@ -0,0 +1,27 @@
+# Precompile
+
+The `precompile` module defines stateful precompile contracts used for creating precompile contracts that enhance the EVM, enable interactions between cosmos state and EVM state, or are used soley for testing purposes.
+
+This module is made of two submodules:
+
+- `contracts` - Defines stateful precompiles and their constructors.
+- `registry` - Defines stateful precompile addresses and registers them with the global registry
+ defined at `github.com/kava-labs/go-ethereum/precompile/modules`.
```

Comment:
```suggestion
This package is made of two subpackages:

- `contracts` - Defines stateful precompiles and their constructors.
- `registry` - Defines stateful precompile addresses and registers them with the global registry
 defined at `github.com/kava-labs/go-ethereum/precompile/modules`.
```

A "submodule" typically refers to a separate module within a larger project, often managed with a separate go.mod file. Submodules are more about versioning and dependency management. In your structure, unless each of these directories (contracts, registry) has its own go.mod file, they are not submodules.

In Go, a "subpackage" refers to a package that is nested within another package directory. Each directory containing Go source files with a package declaration is considered a package. 

Based on the provided structure, contracts and registry are subpackages of the `/preccompiles` package. They are not submodules unless each has its own go.mod file and is managed independently as a separate Go module.

Given that we have an `/x` directory where subdirectories are referred to as modules, we should be careful not to mix up these terms.

Reply:
I have a habit of referring to any unit of code as a module 😓.  Great suggestion and clarification and ya we should ensure when we refer to `./x` directories, we use the term cosmos module to make it distinct from go modules -- though, cosmos main has now made these independent go modules as well.